### ISSUE TITLE
utils.destroy_lock() fails to unlock

### ISSUE THREAD

**Issue by OPNA2608** [chxrlt/lyrebird#35]
On application shutdown, the lock fails to properly unlock.
```
Traceback (most recent call last):
  File "/nix/store/m26bw7cha9cvnk4c0vy2767s1kxir1dv-lyrebird-unstable-2020-08-19/share/lyrebird/lyrebird/mainwindow.py", line 271, in close
    utils.destroy_lock()
  File "/nix/store/m26bw7cha9cvnk4c0vy2767s1kxir1dv-lyrebird-unstable-2020-08-19/share/lyrebird/lyrebird/core/utils.py", line 101, in destroy_lock
    lock_file_path.unlink(missing_ok=True)
TypeError: unlink() got an unexpected keyword argument 'missing_ok'
```

`~/.lyrebird.lock` persists, application is stuck & needs to be killed manually.

---

> ```py
> lock_file_path = Path(Path.home() / '.lyrebird.lock')
> ```

Wouldn't it be better to put the lockfile into the system's tmpdir btw?

**Comment by OPNA2608** [comment]
Fixed by bumping Python to 3.8, please either update the requirements or fix the behaviour.

**Comment by harrego** [comment]
Tomorrow I'll take a closer look at this, I appreciate that you put in this issue!

In the mean time for anyone experiencing this make sure to download the latest stable version (v1.0.2) by following the instructions in the README. The master branch is an unstable experimental branch gearing up to release v1.1.0.

**Comment by harrego** [comment]
> Changed in version 3.8: The missing_ok parameter was added.

Replacing `missing_ok` with a wrapping `lock_file_path.exists()` would bring support back down to Python 3.7.

**Comment by harrego** [comment]
Should be fixed in commit 0fda5ae. If the issue still persists for anyone please let me know.
